lib/LedgerSMB/Database/Change.pm

```diff
@@ -256,14 +254,13 @@ Returns true if the tracking system needs to be initialized
 =cut
 
 sub needs_init {
-    my ($dbh) = @_;
-    local $@ = undef;
-    my $rows = eval { $dbh->prepare(
-       'select 1 from db_patches'
-    )->execute(); };
-    $dbh->rollback;
-    return 0 if $rows;
-    return 1;
+    my $dbh = pop @_;
```

Comment:
This is highly unusual idiom. Why not use `my ($self, $dbh) = @_;` ? or leave out `$self` if that's not part of the deal.